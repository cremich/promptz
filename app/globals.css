pre,
code {
  word-wrap: break-word !important;
  white-space: pre-wrap !important;
  white-space: -moz-pre-wrap !important;
  white-space: -pre-wrap !important;
  white-space: -o-pre-wrap !important;
  margin: 0;
  padding: 0;
}

code {
  background-color: #f3f3f7;
}

pre > code {
  display: inline-block;
  padding: 1em;
}
