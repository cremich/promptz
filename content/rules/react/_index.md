---
title: "React"
description: "Rules for React development, component patterns, and best practices."
---
