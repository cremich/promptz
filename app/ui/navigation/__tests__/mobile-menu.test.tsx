import { describe, expect, test } from "@jest/globals";
import "@testing-library/jest-dom";
import { act, fireEvent, render, screen } from "@testing-library/react";
import MobileMenu from "@/app/ui/navigation/mobile-menu";

describe("Mobile Menu Navigation ", () => {
  test("renders mobile menu unchanged", () => {
    const { container } = render(<MobileMenu />);
    expect(container).toMatchSnapshot();
  });

  test("changes value when clicked", async () => {
    render(<MobileMenu />);

    // get a hold of the button element, and trigger some clicks on it
    const button = screen.getByTestId("menu-button");

    act(() => {
      fireEvent.click(button);
    });

    expect(screen.getByTestId("menu-button-x")).toBeInTheDocument();
  });
});
