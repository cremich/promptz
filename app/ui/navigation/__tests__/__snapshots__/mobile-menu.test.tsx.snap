// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile Menu Navigation  renders mobile menu unchanged 1`] = `
<div>
  <div
    class="md:hidden"
  >
    <button
      class="gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground h-9 w-9 inline-flex items-center justify-center"
      data-testid="menu-button"
    >
      <svg
        class="lucide lucide-menu h-6 w-6"
        data-testid="menu-button-burger"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="4"
          x2="20"
          y1="12"
          y2="12"
        />
        <line
          x1="4"
          x2="20"
          y1="6"
          y2="6"
        />
        <line
          x1="4"
          x2="20"
          y1="18"
          y2="18"
        />
      </svg>
    </button>
  </div>
</div>
`;
